### ISSUE TITLE
Environment variables for service host and port are not created when connect service upstream annotation is used

### ISSUE THREAD

**Issue by mmDonuts** [hashicorp/consul#10794]
#### Overview of the Issue

Some services being deployed (named consumer for clarity) subscribe to events from another service (named producer for clarity). This subscription includes the kubernetes service name in the URL to send events to. Previously the X_CONNECT_SERVICE_HOST environment variable present on the producer pod was used to rewrite the kubernetes service name in the subscription URL to the local proxy endpoint port and localhost. It does not appear that the most recent versions of consul connect are creating those environment variable entries.

#### Reproduction Steps

Expected behavior steps:
1. Create a cluster with previous helm release tied to consul 1.9.x (0.31.1 was used in my testing)
1. Deploy any kubernetes deployment/pod that declares explicit upstream using ```consul.hashicorp.com/connect-service-upstreams:``` annotation
1. View the pod created with that deployment with ```kubectl describe pod ...```
1. _CONNECT_SERVICE_HOST and _CONNECT_SERVICE_PORT environment variables are listed in the output for each upstream

Observed behavior:
1. Create a cluster with current helm release tied to 1.10.x of (0.32.1 was used in my testing)
1. Deploy any service that declares explicit upstream using ```consul.hashicorp.com/connect-service-upstreams:``` annotation
1. View the pod created with that deployment with ```kubectl describe pod ...```
1. _CONNECT_SERVICE_HOST and _CONNECT_SERVICE_PORT environment variables are not listed in the output for each upstream

### Consul info for both Client and Server

<details>
  <summary>Client info</summary>

```
agent:
        check_monitors = 0
        check_ttls = 1
        checks = 3
        services = 2
build:
        prerelease = 
        revision = 27de64da
        version = 1.10.0
consul:
        acl = enabled
        known_servers = 1
        server = false
runtime:
        arch = amd64
        cpu_count = 16
        goroutines = 87
        max_procs = 16
        os = linux
        version = go1.16.5
serf_lan:
        coordinate_resets = 0
        encrypted = false
        event_queue = 0
        event_time = 2
        failed = 0
        health_score = 0
        intent_queue = 0
        left = 0
        member_time = 3
        members = 2
        query_queue = 0
        query_time = 1
```

</details>

<details>
  <summary>Server info</summary>

```
agent:
        check_monitors = 0
        check_ttls = 0
        checks = 0
        services = 0
build:
        prerelease = 
        revision = 27de64da
        version = 1.10.0
consul:
        acl = enabled
        bootstrap = true
        known_datacenters = 1
        leader = true
        leader_addr = 10.42.0.18:8300
        server = true
raft:
        applied_index = 124
        commit_index = 124
        fsm_pending = 0
        last_contact = 0
        last_log_index = 124
        last_log_term = 2
        last_snapshot_index = 0
        last_snapshot_term = 0
        latest_configuration = [{Suffrage:Voter ID:69ddd406-8109-9aea-25bf-91705fc8375d Address:10.42.0.18:8300}]
        latest_configuration_index = 0
        num_peers = 0
        protocol_version = 3
        protocol_version_max = 3
        protocol_version_min = 0
        snapshot_version_max = 1
        snapshot_version_min = 0
        state = Leader
        term = 2
runtime:
        arch = amd64
        cpu_count = 16
        goroutines = 135
        max_procs = 16
        os = linux
        version = go1.16.5
serf_lan:
        coordinate_resets = 0
        encrypted = false
        event_queue = 0
        event_time = 2
        failed = 0
        health_score = 0
        intent_queue = 0
        left = 0
        member_time = 3
        members = 2
        query_queue = 0
        query_time = 1
serf_wan:
        coordinate_resets = 0
        encrypted = false
        event_queue = 0
        event_time = 1
        failed = 0
        health_score = 0
        intent_queue = 0
        left = 0
        member_time = 1
        members = 1
        query_queue = 0
        query_time = 1
```

</details>

### Operating system and Environment details

The info attached here is from a k3d cluster running on Ubuntu 21.04

### Log Fragments

N/A

**Comment by blake** [comment]
Hi @mmDonuts, this issue was also reported in hashicorp/consul-k8s#549 and fixed by PR hashicorp/consul-k8s#550. This fix will be available in the next release of consul-k8s.

**Comment by mmDonuts** [comment]
My apologies for not finding that issue, you can close this one as a duplicate, thanks!
